test/actor/ns-gen/main.go

```diff
@@ -0,0 +1,285 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const nsTemplate = `apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{.namespace}}
+`
+
+const brTemplate = `apiVersion: eventing.knative.dev/v1beta1
+kind: Broker
+metadata:
+  name: testbroker
+  namespace: {{.namespace}}
+  annotations:
+    "eventing.knative.dev/broker.class": "{{.brclass}}"
+`
+
+const envTemplate = `        - name: {{.envname}}
+          value: {{.envvalue}}
+`
+const trTemplate = `apiVersion: v1
+kind: Service
+metadata:
+  name: actor-{{.index}}
+  namespace: {{.namespace}}
+spec:
+  selector:
+    app: actor-{{.index}}
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 8080
+      name: http
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: actor-{{.index}}
+  namespace: {{.namespace}}
+  labels:
+    app: actor-{{.index}}
+spec:
+  replicas: {{.replicas}}
+  selector:
+    matchLabels:
+      app: actor-{{.index}}
+  template:
+    metadata:
+      labels:
+        app: actor-{{.index}}
+    spec:
+      containers:
+      - name: actor
+        image: ko://github.com/google/knative-gcp/test/actor/actor
+        resources:
+          requests:
+            memory: "400Mi"
+          limits:
+            memory: "400Mi"
+        ports:
+        - containerPort: 8080
+        env:
+        - name: AGGREGATOR_ADDR
+          value: http://aggregator.{{.namespace}}.svc.cluster.local
+{{.envs}}
+---
+apiVersion: eventing.knative.dev/v1beta1
+kind: Trigger
+metadata:
+  name: trigger-actor-{{.index}}
+  namespace: {{.namespace}}
+spec:
+  broker: testbroker
+  filter:
+    attributes:
+      type: seed
+  subscriber:
+    ref:
+     apiVersion: v1
+     kind: Service
+     name: actor-{{.index}}
+---
+`
+const seederTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: seeder
+  namespace: {{.namespace}}
+  labels:
+    app: seeder
+spec:
+  replicas: {{.replicas}}
+  selector:
+    matchLabels:
+      app: seeder
+  template:
+    metadata:
+      labels:
+        app: seeder
+    spec:
+      containers:
+      - name: seeder
+        image: ko://github.com/google/knative-gcp/test/actor/seeder
+        env:
+        - name: TARGET
+          value: http://default-brokercell-ingress.cloud-run-events.svc.cluster.local/{{.namespace}}/testbroker
```

Comment:
Do we need to hardcode the broker address?

Reply:
I can see the potential concern but I would rather change it once we have the need to do that.